File: android/app/src/main/java/com/desklink/android/domain/model/ConnectionError.kt
package com.desklink.android.domain.model

enum class ConnectionError(val code: Int, val description: String) {
    REFUSED(1000, "Server refused connection"),
    PROTOCOL_MISMATCH(1001, "Protocol version mismatch"),
    TIMEOUT(1002, "Connection timed out"),
    LOST(1003, "Connection lost"),
    PAIRING_REJECTED(1004, "Pairing rejected (wrong PIN)"),

    ENCODER_INIT_FAILED(1100, "Encoder initialization failed"),
    ENCODER_FAILED(1101, "Encoding error"),
    DECODER_INIT_FAILED(1102, "Decoder initialization failed"),
    DECODER_FAILED(1103, "Decoding error"),
    CODEC_NOT_SUPPORTED(1104, "Codec not supported"),

    DISPLAY_CREATE_FAILED(1200, "Virtual display creation failed"),
    DISPLAY_CAPTURE_FAILED(1201, "Screen capture failed"),
    DISPLAY_RESOLUTION_INVALID(1202, "Invalid resolution"),

    INPUT_INJECTION_FAILED(1300, "Input injection failed"),
    INPUT_PERMISSION_DENIED(1301, "Input permission denied"),

    CONFIG_INVALID(1400, "Invalid configuration"),
    CONFIG_NEGOTIATION_FAILED(1401, "Config negotiation failed");

    companion object {
        fun fromCode(code: Int): ConnectionError? = entries.find { it.code == code }
    }
}

File: macos/DeskLink/Sources/Domain/Models/ConnectionError.swift
import Foundation

public enum ConnectionError: Int, Error, Sendable {
    case refused = 1000
    case protocolMismatch = 1001
    case timeout = 1002
    case lost = 1003
    // The client has had this code all along; the Mac could not send it, so a wrong PIN was
    // signalled by silence -- which the client reads as an unreachable Mac.
    case pairingRejected = 1004
    // Distinct from a rejection: the code may be right and waiting is what helps.
    case pairingLockedOut = 1005

    case encoderInitFailed = 1100
    case encoderFailed = 1101
    case decoderInitFailed = 1102
    case decoderFailed = 1103
    case codecNotSupported = 1104

    case displayCreateFailed = 1200
    case displayCaptureFailed = 1201
    case displayResolutionInvalid = 1202

    case inputInjectionFailed = 1300
    case inputPermissionDenied = 1301

    case configInvalid = 1400
    case configNegotiationFailed = 1401
}
